plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -232,6 +232,64 @@ public void testShowColumns()
         assertEquals(expected, actual);
     }
 
+    @Test
+    public void testNullPredicate()
+            throws IOException
+    {
+        String indexName = "null_predicate1";
+        @Language("JSON")
+        String properties = "" +
+            "{" +
+            "  \"properties\":{" +
+            "    \"null_keyword\":   { \"type\": \"keyword\" }," +
+            "    \"custkey\":   { \"type\": \"keyword\" }" +
+            "  }" +
+            "}";
+        createIndex(indexName, properties);
+        index(indexName, ImmutableMap.<String, Object>builder()
+            .put("null_keyword", 32)
+            .put("custkey", 1301)
+            .build());
+
+        assertQueryReturnsEmptyResult("SELECT * FROM null_predicate1 WHERE null_keyword IS NULL");
+        assertQueryReturnsEmptyResult("SELECT * FROM null_predicate1 WHERE null_keyword = '10' OR null_keyword IS NULL");
+
+        assertQuery("SELECT custkey, null_keyword FROM null_predicate1 WHERE null_keyword = '32' OR null_keyword IS NULL", "VALUES (1301, 32)");
+        assertQuery("SELECT custkey FROM null_predicate1 WHERE null_keyword = '32' OR null_keyword IS NULL", "VALUES (1301)");
+
+        // not null filter
+        // filtered column is selected
+        assertQuery("SELECT custkey, null_keyword FROM null_predicate1 WHERE null_keyword IS NOT NULL", "VALUES (1301, 32)");
+        assertQuery("SELECT custkey, null_keyword FROM null_predicate1 WHERE null_keyword = '32' OR null_keyword IS NOT NULL", "VALUES (1301, 32)");
+
+        // filtered column is not selected
+        assertQuery("SELECT custkey FROM null_predicate1 WHERE null_keyword = '32' OR null_keyword IS NOT NULL", "VALUES (1301)");
+
+        indexName = "null_predicate2";
+        properties = "" +
+            "{" +
+            "  \"properties\":{" +
+            "    \"null_keyword\":   { \"type\": \"keyword\" }," +
+            "    \"custkey\":   { \"type\": \"keyword\" }" +
+            "  }" +
+            "}";
+        createIndex(indexName, properties);
+        index(indexName, ImmutableMap.<String, Object>builder()
+            .put("custkey", 1301)
+            .build());
+
+        // not null filter
+        assertQueryReturnsEmptyResult("SELECT * FROM null_predicate2 WHERE null_keyword IS NOT NULL");
+        assertQueryReturnsEmptyResult("SELECT * FROM null_predicate2 WHERE null_keyword = '10' OR null_keyword IS NOT NULL");
+
+        // filtered column is selected
+        assertQuery("SELECT custkey, null_keyword FROM null_predicate2 WHERE null_keyword IS NULL", "VALUES (1301, NULL)");
+        assertQuery("SELECT custkey, null_keyword FROM null_predicate2 WHERE null_keyword = '32' OR null_keyword IS NULL", "VALUES (1301, NULL)");
```

Comment:
We should also have a document in the index that has a non-null value for field `null_keyword` (say '32'), as well, and show that a filter like `null_keyword = '32' OR null_keyword IS NULL` selects both the null and non-null values (that was the original issue reported).

Reply:
Added.